tedana/viz.py

```diff
@@ -0,0 +1,257 @@
+        ax_fft.plot(freqs, spectrum)
+        ax_fft.set_title('One Sided fft')
+        ax_fft.set_xlabel('Hz')
+        ax_fft.set_xbound(freqs[0], freqs[-1])
+
+        # Fix spacing so TR label does overlap with other plots
+        allplot.subplots_adjust(hspace=0.4)
+        plot_name = 'comp_{}.png'.format(str(compnum).zfill(3))
+        compplot_name = os.path.join('figures', plot_name)
+        plt.savefig(compplot_name)
+        plt.close()
+
+
+def writekappascatter(comptable):
+    """
+    Creates a scatter plot of Kappa vs Rho values. The shape and size of the
+    points is based on classification and variance explained, respectively.
+
+    Parameters
+    ----------
+    comptable : (N x 5) array_like
+        Array with columns denoting (1) index of component, (2) Kappa score of
+        component, (3) Rho score of component, (4) variance explained by
+        component, and (5) normalized variance explained by component
+
+    """
+
+    # Creating Kappa Vs Rho plot
+    ax_scatter = plt.gca()
+
+    # Prebuild legend so that the marker sizes are uniform
+    plt.scatter([], [], s=1, marker='*', c='g', label='accepted', alpha=0.5)
+    plt.scatter([], [], s=1, marker='v', c='r', label='rejected', alpha=0.5)
+    plt.scatter([], [], s=1, marker='d', c='k', label='ignored', alpha=0.5)
+    plt.scatter([], [], s=1, marker='^', c='m', label='midk', alpha=0.5)
+    ax_scatter.legend(markerscale=10)
+
+    # Set up for varying marker shape and color
+    mkr_dict = {'accepted': '*', 'rejected': 'v', 'ignored': 'd', 'midk': '^'}
+    col_dict = {'accepted': 'g', 'rejected': 'r', 'ignored': 'k', 'midk': 'm'}
+    for kind in mkr_dict:
+        d = comptable[comptable.classification == kind]
+        plt.scatter(d.kappa, d.rho,
+                    s=150 * d['variance explained'], marker=mkr_dict[kind],
+                    c=col_dict[kind], alpha=0.5)
+
+    # Finish labeling the plot.
+    ax_scatter.set_xlabel('kappa')
+    ax_scatter.set_ylabel('rho')
+    ax_scatter.set_title('Kappa vs Rho')
+    ax_scatter.xaxis.label.set_fontsize(20)
+    ax_scatter.yaxis.label.set_fontsize(20)
+    ax_scatter.title.set_fontsize(25)
+    scatter_title = os.path.join('figures', 'Kappa_vs_Rho_Scatter.png')
+    plt.savefig(scatter_title)
+
+    plt.close()
+
+
+def writesummaryfig(comptable):
+    """
+    Creates a bar graph showing total variance explained by each component
+    as well as the number of components identified for each category.
+
+    Parameters
+    ----------
+    comptable : (N x 5) array_like
+        Array with columns denoting (1) index of component, (2) Kappa score of
+        component, (3) Rho score of component, (4) variance explained by
+        component, and (5) normalized variance explained by component
+    """
+
+    # Get the variance associated with each classification
+    var_acc = np.sum(comptable[comptable.classification == 'accepted']['variance explained'])
```

Comment:
What if we update these two blocks like this:

```suggestion
var_expl = []
counts = []

for clf in ['accepted', 'rejected', 'ignored']:
    var_expl += np.sum(comptable[comptable.classification == clf]['variance explained'])
    counts += comptable[comptable.classification == clf].count()[0]
```

Then you can just index the lists below.

Reply:
Thanks for this, I  used this framework, but made variance explained a list, counts a dict. just a little bit of knowledge really does seem to be a force multiplier